bin/flutter

```diff
@@ -127,7 +127,11 @@ function upgrade_flutter () {
 
     retry_upgrade
 
-    "$DART" $FLUTTER_TOOL_ARGS --snapshot="$SNAPSHOT_PATH" --packages="$FLUTTER_TOOLS_DIR/.packages" "$SCRIPT_PATH"
+    if [[ -z "$FLUTTER_AOT_TOOL" ]]; then
+      "$DART" $FLUTTER_TOOL_ARGS --snapshot="$SNAPSHOT_PATH" --packages="$FLUTTER_TOOLS_DIR/.packages" "$SCRIPT_PATH"
+    else
+      "$DART_2_NATIVE" aot --packages="$FLUTTER_TOOLS_DIR/.packages" "$SCRIPT_PATH" "$SNAPSHOT_PATH"
```

Comment:
Why isn't this just `dart2aot`?

Reply:
?

Reply:
https://dart.dev/tools/dart2aot

Reply:
Using an env var to locate it is fine, but I don't think that command takes an aot argument.

Reply:
I mean, it does, that's how I tested it :)

Reply:
Hmm strange.  Ok.

Reply:
should we also check that the binary at `$DART_2_NATIVE` exists? or are we assuming that if you set the `FLUTTER_AOT_TOOL` var you know what you're doing?

Reply:
To some extent, this is a "you know what you are doing" feature for testing/experimentation.